utils/streams.js

```
@@ -0,0 +1,220 @@
+
+        callback();
+      },
+      function end() {
+        this.push(']'); // At the last step we close array bracket
+      },
+    ))
+    .pipe(fs.createWriteStream(`${DIR_PATH}/${CONVERTED_FILE_NAME}`))
+    .on('error', handleError);
+};
+
+
+const cssBundler = (path) => {
+  if (isDirectoryNotExist(path)) {
+    return handleError('This file does not exist. Maybe you forget to pass the --path(-p) option?');
+  }
+
+  const END_STYLES_FILE_NAME = 'end-styles.css';
+  const BUNDLE_FILE_NAME = 'bundle.css';
+  const CSS_REGEX = /^.*.css$/g;
+
+  const readStreams = [];
+  const endStylesStream = fs.createReadStream(`${path}/${END_STYLES_FILE_NAME}`);
+
+  fs.readdirSync(path).forEach((fileName) => {
+    if (CSS_REGEX.test(fileName) && fileName !== END_STYLES_FILE_NAME) {
+      readStreams.push(fs.createReadStream(`${path}/${fileName}`));
+    }
+
+    const combinedStream = CombinedStream.create();
+
+    readStreams.forEach(readStream => combinedStream.append(readStream));
+
+    combinedStream.append(endStylesStream);
+
+    combinedStream
+      .pipe(fs.createWriteStream(`${path}/${BUNDLE_FILE_NAME}`))
+      .on('error', handleError);
+  });
+};
+
+
+const commandExecutor = (action, path) => {
+  switch (action) {
+    case ACTIONS.REVERSE:
+      return reverse();
+    case ACTIONS.TRANSFORM:
+      return transform();
+    case ACTIONS.OUTPUT_FILE:
+      return outputFile(path);
+    case ACTIONS.CONVERT_FROM_FILE:
+      return convertFromFile(path);
+    case ACTIONS.CONVERT_TO_FILE:
+      return convertToFile(path);
+    case ACTIONS.CSS_BUNDLER:
+      return cssBundler(path);
+    default:
+      handleError('This action does not exist.');
+      return program.help();
+  }
+};
+
+
+program
+  .option('-a, --action', 'Name of the called action. - required')
+  .option('-f, --file', 'The path to the file being used.')
+  .option('-p, --path', 'Grab all css files from the given path provided by --path option')
+  .action(commandExecutor)
```

The problem is that arguments are passed to the function as array and there is no way to get to know a key a specific value belongs to. So I can specify **-a** along with **-p** for convertFromFile action and it will pass (but should throw an error instead as **-f** key is not specified).